vendor/bat-native-ledger/src/bat/ledger/internal/common/time_util.cc

```diff
@@ -33,4 +35,23 @@ uint64_t GetCurrentTimeStamp() {
   return static_cast<uint64_t>(base::Time::Now().ToDoubleT());
 }
 
+base::TimeDelta GetRandomizedDelay(base::TimeDelta delay) {
+  uint64_t seconds = brave_base::random::Geometric(delay.InSecondsF());
+  return base::TimeDelta::FromSeconds(static_cast<int64_t>(seconds));
+}
+
+base::TimeDelta GetRandomizedDelayWithBackoff(
+    base::TimeDelta delay,
+    base::TimeDelta max_delay,
+    int backoff_count) {
+  for (int i = 0; i < backoff_count; ++i) {
+    delay *= 2;
+    if (delay > max_delay) {
+      delay = max_delay;
+      break;
+    }
+  }
```

Comment:
`delay` here is a `TimeDelta`, so unfortunately that won't work. I recommend that we try to use `TimeDelta`s where possible, since they have things like overflow checking built-in, and their methods have readable names.

As far as this loop goes, I suspect that we'll have other problems before `backoff_count` gets so large that performance becomes an issue.